app/models/highlight_suggestion.rb

```diff
@@ -0,0 +1,68 @@
+class HighlightSuggestion < ApplicationRecord
+  belongs_to :run
+
+  # hunt looks on Twitch for past broadcasts whose timestamps imply that it contains the run's PB. If found, the
+  # highlight suggestion is saved to the database. If not, it is not.
+  def hunt!
+    return if run.user.nil? || run.user.twitch.nil?
+
+    pb = run.histories.find_by(
+      realtime_duration_ms: run.duration_ms(Run::REAL),
+      gametime_duration_ms: run.duration_ms(Run::GAME)
+    )
+    return if pb.nil?
+
+    Twitch::Videos.recent(run.user.twitch.twitch_id).each do |video|
+      match = /^((\d+)h)?((\d+)m)?((\d+)s)?$/.match(video['duration'])
+      hours   = match[2].to_i.hours
+      minutes = match[4].to_i.minutes
+      seconds = match[6].to_i.seconds
+
+      video_start = DateTime.parse(video['created_at'])
+      video_end   = video_start + hours + minutes + seconds
+
+      next unless video['type'] == 'archive'
```

Comment:
Instead of getting all videos, why not just select that we only want archives from helix?  The `type` parameter will accept `archive` which should get rid of uploads and highlights.